src/include/migraphx/algorithm.hpp

```diff
@@ -54,6 +55,25 @@ T transform_accumulate(Iterator first, Iterator last, T init, BinaryOp binop, Un
         first, last, first, init, binop, [&](auto&& x, auto&&) { return unaryop(x); });
 }
 
+template <class Iterator, class OutputIterator, class BinaryOperation, class UnaryOp>
+OutputIterator transform_partial_sum(
+    Iterator first, Iterator last, OutputIterator d_first, BinaryOperation binop, UnaryOp unaryop)
```

Comment:
Add docstring. Ideally for such functions, better if you can demonstrate example usecase in docstring.